io/castor/CMakeLists.txt

```diff
@@ -8,10 +8,12 @@ if(CASTOR_VERSION VERSION_GREATER "2.0.0")
   add_definitions(-DR__CASTOR2)
 endif()
 
-ROOT_ADD_CXX_FLAG(CMAKE_CXX_FLAGS -Wno-shadow)
-
 ROOT_STANDARD_LIBRARY_PACKAGE(RCastor
+                              HEADERS TCastorFile.h
+                              SOURCES src/TCastorFile.cxx
```

Comment:
This is fine, but I believe adding the `src/` part is not necessary.

Reply:
Indeed - `ROOT_STANDARD_LIBRARY_PACKAGE` will try both with and without `src/` -- a piece of magic I really don't like :(

Seems requiring internal users be consistent would be useful here.

Reply:
Well, it's confusing to add full path for sources and not for headers. I think it's more consistent to omit it in both cases. We will eventually remove most "magic", but we cannot do it without renaming the macro, since now it became an interface used by external projects.

Reply:
For the sake of moving forward I will merge this as is and we can revisit the style later.